### DISCUSSION TITLE
Swapping a plain-text mask for an encryption method

### DISCUSSION THREAD

**Discussion by addijhaq** [logfellow/logstash-logback-encoder#1048] [General]
Hi, I was wondering if there was a simple way to be able to mask values using an encryption method as opposed to a plain text mask. 

if not, how would i go about implementing such a feature?

**Comment by philsttr** [comment]
You can implement [`FieldMasker`](https://github.com/logfellow/logstash-logback-encoder/blob/main/src/main/java/net/logstash/logback/mask/FieldMasker.java) or [`ValueMasker`](https://github.com/logfellow/logstash-logback-encoder/blob/main/src/main/java/net/logstash/logback/mask/ValueMasker.java)

And then specify either in the `MaskingJsonGeneratorDecorator`

```xml
<decorator class="net.logstash.logback.mask.MaskingJsonGeneratorDecorator">
    <fieldMasker class="your.custom.EncryptingFieldMasker"/>
    <valueMasker class="your.custom.EncryptingValueMasker"/>
</decorator>
```

https://github.com/logfellow/logstash-logback-encoder?tab=readme-ov-file#masking
